### ISSUE TITLE
class.comparison Raster error

### ISSUE THREAD

**Issue by Saadi4469** [jeffreyevans/spatialEco#31]
Hello,

I am trying to use the `class.comparison` function to perform `t.test` on two landcover classification rasters in `GeoTiff` format. Now the `?class.comparison` tells me that  the function does support raster objects, but it's still giving me the error below. How can I fix this? Or do I have to convert the raster into a `SpatialPixelsDataFrame`?

Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘gridded’ for signature ‘"RasterLayer"’

**Comment by jeffreyevans** [comment]
The class.comparison is a bit of a legacy function that is kept around because of the implementation of the sub-sampling option and to support sp class objects. I believe that you are wanting the raster.change function, which supports raster class objects (and terra in the next release of spatialEco). This function also has the kappa and t.test along with correlation, entropy, cross-entropy and divergence options for the statistical comparison. I will add something to the help as well as a warning message to this effect to avoid confusion in the future.

Best,
Jeff

Jeffrey S. Evans, Ph.D., | Senior Landscape Ecologist & Biometrician
The Nature Conservancy | Protected Lands Science
Visiting Professor | University of Wyoming | Zoology & Physiology
Laramie, WY | ***@***.******@***.***> | (970) 672-6766<tel:(970)%20672-6766>

From: Saad Baloch ***@***.***>
Sent: Thursday, July 22, 2021 5:13 PM
To: jeffreyevans/spatialEco ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jeffreyevans/spatialEco] class.comparison Raster error (#31)


Hello,

I am trying to use the class.comparison function to perform t.test on two landcover classification rasters in GeoTiff format. Now the ?class.comparison tells me that the function does support raster objects, but it's still giving me the error below. How can I fix this? Or do I have to convert the raster into a SpatialPixelsDataFrame?

Error in (function (classes, fdef, mtable) :
unable to find an inherited method for function ‘gridded’ for signature ‘"RasterLayer"’

**Comment by Saadi4469** [comment]
So, should I instead use the `raster.change` function? Currently, `R` is running the `class.comparison` function for the `kappa` test, and it has been running for hours now (pixel resolution is 0.07 m). Is the `raster.change` faster than the legacy function, or will it also take time since the resolution is quite high? 

Thank you,

**Comment by jeffreyevans** [comment]
Yes, the raster.change function would be preferred and uses the raster focal function rather than an iterated vector neighbor index so, will be considerably faster. However, 0.07m is very high resolution is will take some time regardless. I will push the new version of the function to GitHub in the next few days so that there will be terra support which really speeds up the focal function via Rcpp. But, it will still take time at this resolution.

-Jeff
